### PR TITLE
fix: rework host shutdown

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

- Always include a timeout for graceful shutdown (e.g. if NATS connection dies, it will never finish)
- Stop if one of the core wasmbus tasks dies
- Flush NATS queues concurrently on shutdown
- Handle `stopped` method errors

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Manual Verification

build and run, then:
- try starting the host, killing NATS and `^C` - it will shutdown after 10 seconds of trying to gracefully shutdown
- the usual `wash stop host`

### CODE REVIEW

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -1921,13 +1921,20 @@ impl Host {
...
+                    Abortable::new(queue, queue_abort_reg)
+                        .for_each(move |msg| {
+                            let host = Arc::clone(&host);
+                            async move { host.handle_message(msg).await }
+                        })
+                        .await;
+                }
+                let deadline = { *host.stop_rx.borrow() };
+                host.stop_tx.send_replace(deadline);
+                info!("control interface queue task stopped");
```
Comment (thomastaylor312) [inline crates/host/src/wasmbus/mod.rs:1937]: Nit: Will we only ever get here (and in the other tasks) on clean exit or could we also get here if the task aborts early? If we can abort early, should this be a `warn` or `error`?

**Review by thomastaylor312 — DISMISSED** [review]
Just one small nit here

File: crates/host/src/wasmbus/mod.rs (hunk 1 above)
Comment (rvolosatovs) [reply crates/host/src/wasmbus/mod.rs:1937]: both and IMO it'd be pretty confusing to get a `warn`/`error` on clean exit

Comment (rvolosatovs) [reply crates/host/src/wasmbus/mod.rs:1937]: We will get an error before that from the actual stream though

Comment (connorsmith256) [reply crates/host/src/wasmbus/mod.rs:1937]: To clarify: I'm wondering whether there's a case where the underlying NATS client's pollable returns a `Poll::Ready(None)` without us intentionally unsubscribing

Comment (rvolosatovs) [reply crates/host/src/wasmbus/mod.rs:1937]: > I have the same question, but I think the answer is: this and the other tasks below can be stopped from receiving a signal (ctrl-C, or a `.stop` command on the control interface), or abnormally if the underlying stream finishes (which shouldn't ever happen). @rvolosatovs is there a way to handle when the stream unexpectedly closes?

did not see this before answering.
we can actually just check https://docs.rs/futures/latest/futures/stream/struct.Abortable.html#method.is_aborted
It won't always be true, there's a race condition, but it should be good enough - let me do that.

Comment (connorsmith256) [reply crates/host/src/wasmbus/mod.rs:1937]: Oo I like that idea 👍

**Review by connorsmith256 — APPROVED** [review]

**Review by thomastaylor312 — APPROVED** [review]
Looking great!

**Pull request merged**